### PR TITLE
DPL: Using InputSpan as argument for CompletionPolicy callback

### PR DESCRIPTION
**Pull request by matthiasrichter**
The change is transparent for the user code as the InputSpan provides an
iterator interface. Using InputSpan will allow to have access to the parts
within one route in order to create more complex completion checks.

### CODE REVIEW

File: Framework/Core/include/Framework/InputSpan.h [hunk 1]
```diff
@@ -10,6 +10,8 @@
 #ifndef FRAMEWORK_INPUTSPAN_H
 #define FRAMEWORK_INPUTSPAN_H
 
+#include <functional>
+
```
Comment (ktf) [inline Framework/Core/include/Framework/InputSpan.h]: Do you really need functional in the header? It's an heavy beast.

Comment (matthiasrichter) [reply Framework/Core/include/Framework/InputSpan.h]: Actually i though we need this to use `std::function`, and that it was simply missing. `InputSpan.h` would then have hidden dependencies. To my surprise `std::function` can be used without functional. But I found that `DataRef.h` is missing. Will change this.

Comment (matthiasrichter) [reply Framework/Core/include/Framework/InputSpan.h]: Did a test with just including `InputSpan.h`, `functional is indeed a hidden dependency, so I would add it.

Comment (matthiasrichter) [reply Framework/Core/include/Framework/InputSpan.h]: I have now also changed the order of include files in the test to include `InputSpan.h` first and by that check for hidden dependencies.

**Pull request merged**